src/txmempool.cpp

```diff
@@ -868,6 +899,19 @@ void CTxMemPool::check(const CCoinsViewCache *pcoins) const
                 assert(pcoins->HaveSidechain(fwd.scId));
         }
 
```

Comment:
We should verify that referenced cert in MBTR is correct, but more in general do all the state related consistency check for all the sc related elements. Use IsSCTxApplicableToState?

Reply:
This proved to be quite more challenging that foreseen and we'd propose to move it to a separate PR.
Main reason is that current `CTxMemPool::check` handles unconfirmed txes/certs in a pretty flat way, while updating sidechains required a delicate balance of ordered txes/certs, like it's done for blocks.
We would like to keep going with integrating the new features and delay updating these checks